core/src/main/java/org/jruby/RubyDir.java

```diff
@@ -624,6 +624,15 @@ public IRubyObject rewind() {
         return this;
     }
 
+    @JRubyMethod(name = "empty?", meta = true)
+    public static IRubyObject empty_p(ThreadContext context, IRubyObject recv, IRubyObject arg) {
+        Ruby runtime = context.runtime;
+        RubyString path = StringSupport.checkEmbeddedNulls(runtime, RubyFile.get_path(context, arg));
+        RubyFileStat fileStat = runtime.newFileStat(path.asJavaString(), false);
+        boolean isDirectory = fileStat.directory_p().isTrue();
```

Comment:
Oh, I wasn't aware you had additional changes planned.

So the problem is that this isn't raising now, yes? You can just raise the error yourself via `throw runtime.newErrnoENOENT(....)`. If there's common logic across a couple methods, that could also be pulled out into something common.